crates/consensus/auto-seal/src/lib.rs

```diff
@@ -449,45 +372,46 @@ impl StorageInner {
 
         trace!(target: "consensus::auto", transactions=?&block.body, "executing transactions");
 
-        // now execute the block
-        let db = State::builder()
-            .with_database_boxed(Box::new(StateProviderDatabase::new(
-                client.latest().map_err(BlockExecutionError::LatestBlock)?,
-            )))
-            .with_bundle_update()
-            .build();
-        let mut executor = EVMProcessor::new_with_state(chain_spec.clone(), db, evm_config);
+        let mut db = StateProviderDatabase::new(
+            provider.latest().map_err(BlockExecutionError::LatestBlock)?,
+        );
 
-        let (bundle_state, gas_used) = self.execute(&block, &mut executor)?;
+        // TODO(mattsse): At this point we don't know certain fields of the header, so we first
+        // execute it and then update the header this can be improved by changing the executor
+        // input, for now we intercept the errors and retry
+        loop {
+            match executor.executor(&mut db).execute((&block, U256::ZERO).into()) {
```

Comment:
is this a replacement for `complete_header`

Reply:
yes